ivy/functional/frontends/torch/nn/functional/loss_functions.py

```diff
@@ -303,6 +303,47 @@ def nll_loss(
     return ret
 
 
+@to_ivy_arrays_and_back
+def gaussian_nll_loss(
+    input,
+    target,
+    var,
+    full=False,
+    eps=1e-6,
+    reduction="mean"
+):
+
+    if var.shape != input.shape:
+        if input.shape[:-1] == var.shape:
+            var.unsqueeze(dim=2)
+    elif input.shape[:-1] == var.shape[:-1] and var.shape[-1] == 1:
+        pass
+    else:
+        ivy.utils.exceptions.IvyError(
+            f"var is of incorrect size"
+        )
```

Comment:
I don't think these checks match what the documentation says about the shapes. For example, any one dimension could be equal to 1 not just the last one. Maybe you could remove these check altogether for now and see what errors come up while testing.

Reply:
Yeah there was bug here. All the if-else conditions should be nested under the `if var.shape != input.shape:`. I have fixed it now.